api/src/main/java/org/openmrs/api/impl/LocationServiceImpl.java

```diff
@@ -132,21 +132,18 @@ public Location getDefaultLocation() throws APIException {
 		return location;
 	}
 
-	private Location getDefaultLocation(Location location, String locationGP) {
+	private Location getDefaultLocation(Location location, String locationGP, String defaultLocation) {
 		//Try to look up 'Unknown Location' in case the global property is something else
 		Location result = getDefaultLocationFromSting(location, locationGP, "Unknown Location");
 
 		// If Unknown Location does not exist, try Unknown if the global property was different
 		if (result == null) {
 			result = getDefaultLocationFromSting(location, locationGP, "Unknown");
 		}
-
-		return result;
-	}
-
-	private Location getDefaultLocationFromSting(Location location, String locationGP, String defaultLocation) {
+		
 		Location result = null;
-		if (location == null && (!StringUtils.hasText(locationGP) || !defaultLocation.equalsIgnoreCase(locationGP))) {
+		
+		else if (location == null && (!StringUtils.hasText(locationGP) || !defaultLocation.equalsIgnoreCase(locationGP))) {
```

Comment:
Build is failing because you have a `else if` condition without a `if` condition here